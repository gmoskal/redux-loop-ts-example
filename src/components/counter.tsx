import * as React from "react"
import { connect, Dispatch } from "react-redux"
import { bindActionCreators } from "redux"
import { State, actions } from "../reducers/"

type Props = State & typeof actions

export const Counter: React.SFC<Props> = ({ counter, increment, save, load, reset, isSaving, isLoading, error }) => (
    <div>
        <h1>{isLoading ? "..." : counter}</h1>
        <button disabled={isLoading} onClick={load}>
            {isLoading ? "loading..." : "load"}
        </button>
        <button onClick={increment}>increment</button>
        <button disabled={isSaving} onClick={save}>
            {isSaving ? "saving..." : "save"}
        </button>
        <br />
        <button onClick={reset}>reset</button>
        <h1 className="error">{error ? error.message : null}</h1>
    </div>
)

const mapStateToProps = (state: State): State => state
const mapDispatchToProps = (dispatch: Dispatch<State>): typeof actions => bindActionCreators(actions, dispatch)

export const CounterContainer = connect<State, typeof actions>(mapStateToProps, mapDispatchToProps)(Counter)
