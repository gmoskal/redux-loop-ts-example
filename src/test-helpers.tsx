import * as React from "react"
import * as shallowRenderer from "react-test-renderer/shallow"
import { State, initialState, reducer } from "./reducers/index"
import { Provider } from "react-redux"
import * as deepRenderer from "react-test-renderer"
import { createStore, compose } from "redux"
import { install, StoreCreator } from "redux-loop"

const enhancedCreateStore = createStore as StoreCreator

const enhancer = compose(install())
const store = enhancedCreateStore(reducer, initialState, enhancer)

export const renderComponentShallow = (component: React.ReactElement<any>) =>
    shallowRenderer.createRenderer().render(component)

export const renderComponent = (component: React.ReactElement<any>, _state: State = initialState) =>
    deepRenderer.create(<Provider store={store}>{component}</Provider>)
