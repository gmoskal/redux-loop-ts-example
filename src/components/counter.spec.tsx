import * as React from "react"
import { renderComponent } from "../test-helpers"
import { CounterContainer } from "./counter"

describe("components/Counter", () =>
    it("renders", () => expect(renderComponent(<CounterContainer />)).toMatchSnapshot()))
