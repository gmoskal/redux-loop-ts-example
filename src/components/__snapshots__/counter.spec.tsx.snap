// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Counter renders 1`] = `
<div>
  <h1>
    0
  </h1>
  <button
    onClick={[Function]}
  >
    increment
  </button>
  <br />
  <button
    disabled={undefined}
    onClick={[Function]}
  >
    load
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    disabled={undefined}
    onClick={[Function]}
  >
    save
  </button>
  <h1
    className="error"
  />
</div>
`;
